src/microceph.py

```diff
@@ -27,6 +27,22 @@ def remove_cluster_member(name: str, is_force: bool) -> None:
     _run_cmd(cmd)
 
 
+def get_public_address() -> str:
+    """Returns first mon host address as read from the ceph.conf file."""
+    conf_file_path = "/var/snap/microceph/current/conf/ceph.conf"
+    public_addrs = []
+    with open(conf_file_path, "r") as conf_file:
+        lines = conf_file.readlines()
+        for line in lines:
+            if "mon host" in line:
+                addrs = line.split(" ")[-1].split(",")
+                logger.debug(f"Found public addresses {addrs} in conf file.")
+                public_addrs.extend(addrs)
+                break
+
+    return public_addrs[0]
```

Comment:
This doesn't look right. The radosgw charm, for example, expects every mon unit to send its actual address (not just a random one) so it can have a mirrored view of the ceph.conf file. What we should do, IMO, is get a list of the ceph-mon addresses like we're doing here and then compare it against the unit's public interfaces (using `netifaces`, for example) to see which one corresponds to it.

Reply:
Hmm, Nice catch, I assumed it to be populating relation databag since there were no unit identifiers, but the information updated is unit level indeed.
![Uploading image.png…]()

Reply:
will resolve.